tools/c7n_azure/c7n_azure/arm.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import six
+from c7n_azure.query import QueryResourceManager, QueryMeta
+from c7n_azure.actions import Tag
+from c7n_azure.utils import ResourceIdParser
+
+
+class ArmResourceQueryMeta(QueryMeta):
```

Comment:
Can we replace this class with just `ArmResourceManager.filter_registry.register('tag', Tag)` or something like that?

Reply:
Yeah, I really wanted to. I tried couple of things (e.g putting this line to init or new). The problem is that schema validation happens before any of those functions executed and it fails validation. Let’s talk about that offline.